linker/linker_soinfo.h

```diff
@@ -338,6 +340,40 @@ struct soinfo {
   uintptr_t handle_;
 
   friend soinfo* get_libdl_info(const char* linker_path, const link_map& linker_map);
+
+ public:
+  struct SegmentInfo {
+    ElfW(Addr) phdr_addr, real_addr;
+    ElfW(Word) real_size, page_size;
+    size_t index;
+  };
+
+  typedef std::vector<SegmentInfo> seginfo_list_t;
```

Comment:
Would not create a `typedef` for this, it's only used in 3 places and not that much shorter.
Also `*_list_t` in C-ish code makes me think of a linked list.